src/main/kotlin/org/jitsi/jicofo/bridge/ExternalBridgeSelectionStrategy.kt

```diff
@@ -0,0 +1,110 @@
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+
+@Suppress("unused")
+class ExternalBridgeSelectionStrategy() : BridgeSelectionStrategy() {
+    private val httpClient: HttpClient = HttpClient
+        .newBuilder()
+        .connectTimeout(ExternalBridgeSelectionStrategyConfig.config.timeout)
+        .build()
+
+    private val logger: Logger = LoggerImpl(ExternalBridgeSelectionStrategy::class.simpleName)
+
+    private val fallbackStrategy: BridgeSelectionStrategy? by lazy {
+        val fallbackStrategyName = ExternalBridgeSelectionStrategyConfig.config.fallbackStrategy ?: return@lazy null
+        try {
+            val clazz = Class.forName("${javaClass.getPackage().name}.$fallbackStrategyName")
+            clazz.getConstructor().newInstance() as BridgeSelectionStrategy
+        } catch (e: Exception) {
+            val clazz = Class.forName(fallbackStrategyName)
+            clazz.getConstructor().newInstance() as BridgeSelectionStrategy
+        }
+    }
+
+    private fun fallback(
+        bridges: MutableList<Bridge>?,
+        conferenceBridges: MutableMap<Bridge, Int>?,
+        participantRegion: String?
+    ): Bridge {
+        if (fallbackStrategy == null) {
+            throw Exception("External bridge selection failed and no fallbackStrategy was provided.")
+        }
+        return fallbackStrategy!!.doSelect(bridges, conferenceBridges, participantRegion)
+    }
+
+    override fun doSelect(
+        bridges: MutableList<Bridge>?,
+        conferenceBridges: MutableMap<Bridge, Int>?,
+        participantRegion: String?
+    ): Bridge {
+        val url = ExternalBridgeSelectionStrategyConfig.config.url
+            ?: throw Exception("ExternalBridgeSelectionStrategy requires url to be provided")
+
+        val requestBody = JSONObject()
+        requestBody["bridges"] = bridges?.map {
+            mapOf(
+                "jid" to it.jid.toString(),
+                "version" to it.version,
+                "colibri2" to it.supportsColibri2(),
+                "relay_id" to it.relayId,
+                "region" to it.region,
+                "stress" to it.stress,
+                "operational" to it.isOperational,
+                "graceful_shutdown" to it.isInGracefulShutdown,
+                "draining" to it.isDraining,
+            )
+        }
+        requestBody["conference_bridges"] = conferenceBridges?.mapKeys { it.key.jid.toString() }
+        requestBody["participant_region"] = participantRegion
+        requestBody["fallback_strategy"] = ExternalBridgeSelectionStrategyConfig.config.fallbackStrategy
+
+        val request = HttpRequest
+            .newBuilder()
+            .POST(HttpRequest.BodyPublishers.ofString(requestBody.toJSONString()))
+            .uri(URI.create(url))
+            .headers("Content-Type", "application/json")
+            .timeout(ExternalBridgeSelectionStrategyConfig.config.timeout)
```

Comment:
Will this work as expected when the timeout is `null`? You could either only call `.timeout` if the config has a value, or make the config value required (`by config` instead of `by optionalconfig`) and add a reasonable default value in `reference.conf`

Reply:
Good catch, `.timeout(null)` [will throw](https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.net.http/share/classes/jdk/internal/net/http/HttpRequestBuilderImpl.java#L231). I think making it required with a default in `reference.conf` is sensible; there is no rational reason to want to block forever...